packages/network-controller/src/NetworkController.ts

```diff
@@ -1063,205 +1481,427 @@ export class NetworkController extends BaseController<
+          };
+        }
+
+        networkClientOperations.push({
+          type: 'replace' as const,
+          oldRpcEndpoint: existingRpcEndpointForReplaceWhenChainNotChanged,
+          newRpcEndpoint,
+        });
+      } else {
+        const newRpcEndpoint =
+          newRpcEndpointFields.type === RpcEndpointType.Infura
+            ? newRpcEndpointFields
+            : { ...newRpcEndpointFields, networkClientId: uuidV4() };
+        const networkClientOperation = {
+          type: 'add' as const,
+          rpcEndpoint: newRpcEndpoint,
+        };
+        networkClientOperations.push(networkClientOperation);
+      }
+    }
+
+    for (const existingRpcEndpoint of existingNetworkConfiguration.rpcEndpoints) {
+      if (
+        !networkClientOperations.some((networkClientOperation) => {
+          const otherRpcEndpoint =
+            networkClientOperation.type === 'replace'
+              ? networkClientOperation.oldRpcEndpoint
+              : networkClientOperation.rpcEndpoint;
+          return (
+            otherRpcEndpoint.type === existingRpcEndpoint.type &&
+            otherRpcEndpoint.networkClientId ===
+              existingRpcEndpoint.networkClientId &&
+            otherRpcEndpoint.url === existingRpcEndpoint.url
+          );
+        })
+      ) {
+        const networkClientOperation = {
+          type: 'remove' as const,
+          rpcEndpoint: existingRpcEndpoint,
+        };
+        networkClientOperations.push(networkClientOperation);
+      }
     }
+
+    const updatedNetworkConfiguration =
+      this.#determineNetworkConfigurationToPersist({
+        networkFields: fields,
+        networkClientOperations,
+      });
+
     if (
-      !existingAutoManagedNetworkClient ||
-      shouldDestroyExistingNetworkClient
+      replacementSelectedRpcEndpointIndex === undefined &&
+      !updatedNetworkConfiguration.rpcEndpoints.some((rpcEndpoint) => {
+        return (
+          rpcEndpoint.networkClientId === this.state.selectedNetworkClientId
+        );
+      }) &&
+      !networkClientOperations.some((networkClientOperation) => {
+        return (
+          networkClientOperation.type === 'replace' &&
+          networkClientOperation.oldRpcEndpoint.networkClientId ===
+            this.state.selectedNetworkClientId
+        );
+      })
     ) {
-      customNetworkClientRegistry[networkClientId] =
-        createAutoManagedNetworkClient({
-          type: NetworkClientType.Custom,
-          chainId,
-          rpcUrl,
-          ticker,
-        });
+      throw new Error(
+        // False negative - this is a string.
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        `Could not update network: Cannot update RPC endpoints in such a way that the selected network '${this.state.selectedNetworkClientId}' would be removed without a replacement. Choose a different RPC endpoint as the selected network via the \`replacementSelectedRpcEndpointIndex\` option.`,
```

Comment:
This error is throwing in cases I don't think it should.  Example:

MM is currently on Mainnet, with a `selectedNetworkClientId` pointing to one of that chain's RPC endpoints.  I edit another chain, like changing the name or endpoints of Base network.

This line throws.  But I'm editing a different chain, which shouldn't conflict.   Wondering if I've got a bug somewhere, or this needs to check whether `selectedNetworkClientId` is on the chain being updated.

Reply:
Hmm, you're right. This doesn't seem right to me either, now that I look at it again. I'll take another look today and try to get this fixed quickly.

Reply:
Alright, I've made some changes that should hopefully fix the issue you're seeing.

Reply:
This should be fixed in the latest preview build, but let me know if not.